### PR TITLE
Add hot reload functionality for label component.

### LINKED ISSUE #3467: Hot reload of label (file) doesn't work (DEF-2897)
Description:

Hot reload of .label file seems to print random memory contents instead of the newly specified text.

**Steps to reproduce:**

Create foo.label, add to game object in main collection. Build/launch. Edit foo.label text, Reload Resource.

**Expected result:**

The text is updated appropriately.

**Actual result:**

The text is not updated

### PR DESCRIPTION
**Pull request by ekharkunov**
Add hot reload functionality for label component.
Reread properties from reloaded label resource and mark component to be rehashed.

Fixes #3467

## PR checklist

* [ ] Code
	* [ ] Add engine and/or editor unit tests.
	* [ ] New and changed code follows the overall code style of existing code
	* [ ] Add comments where needed
* [ ] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [ ] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [ ] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [ ] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [ ] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

----------

Example of a well written PR description:

1. Start with the user facing changes. This will end up in the release notes.
1. Add one of the GitHub approved closing keywords
1. Optionally also add the technical changes made. This is information that might help the reviewer. It will not show up in the release notes. Technical changes are identified by a line starting with one of these:
   1. `### Technical changes` 
   1. `Technical changes:`
   2. `Technical notes:`

```
There was a anomaly in the carbon chroniton propeller, introduced in version 8.10.2. This fix will make sure to reset the phaser collector on application startup.

Fixes #1234

### Technical changes
* Pay special attention to line 23 of phaser_collector.clj as it contains some interesting optimizations
* The propeller code was not taking into account a negative phase.
```

### CODE REVIEW

**Comment by AGulev** [comment]
Please link a ticket you solve with this PR

**Comment by ekharkunov** [comment]
> Please link a ticket you solve with this PR

Added!

**Review by AGulev — DISMISSED** [review]
LGTM!

File: engine/gamesys/src/gamesys/components/comp_label.cpp [hunk 1]
```diff
@@ -543,7 +543,21 @@ namespace dmGameSystem
 
     void CompLabelOnReload(const dmGameObject::ComponentOnReloadParams& params)
     {
-        (void)params;
+        LabelResource* resource = (LabelResource*)params.m_Resource;
+        dmGameSystemDDF::LabelDesc* ddf = resource->m_DDF;
+
+        LabelWorld* label_world = (LabelWorld*)params.m_World;
+        LabelComponent* component = &label_world->m_Components.Get(*params.m_UserData);
+        component->m_Size     = Vector3(ddf->m_Size[0], ddf->m_Size[1], ddf->m_Size[2]);
```
Comment (britzl) [inline engine/gamesys/src/gamesys/components/comp_label.cpp:551]: Is there perhaps some value in sharing this code with the code in CompLabelCreate() which also sets the component values from a LabelDesc?

**Review by britzl — COMMENTED** [review]
Looks good but I have a question about sharing code with CompLabelCreate()

File: engine/gamesys/src/gamesys/components/comp_label.cpp (hunk 1 above)
Comment (ekharkunov) [reply engine/gamesys/src/gamesys/components/comp_label.cpp:551]: I think, yes. I'll make additional function with shared code and will reuse it.

**Comment by ekharkunov** [comment]
@britzl @AGulev I've reworked according to review comments. Please, review one more time.

**Review by AGulev — DISMISSED** [review]
Looks good!

**Comment by ekharkunov** [comment]
Fix misprint InitParamtersFromDescription -> InitParam**e**tersFromDescription

**Review by britzl — APPROVED** [review]
Looks good!

**Pull request merged**